clpy/core/core.pyx

```diff
@@ -347,6 +347,10 @@ cdef class ndarray:
             newarray._set_shape_and_strides(self._shape, strides)
         else:
             newarray = ndarray(self.shape, dtype=dtype, order=order)
+
+        if dtype == numpy.bool_:
+            self[clpy.isnan(self)] = True
```

Comment:
Can you explain why this 27dd0d4a9b62ef3eefac8fad041d9a4b1ec6da88 is needed by ClPy but not by CuPy?

Reply:
```nan``` must be ```True``` according to numpy document.
And, numpy and cupy treats ```nan``` as ```True```.
But, in our clpy (at least in titanv machine ), ```nan``` was treated as ```False```.
So, I add this code, to treat ```nan``` as ```true``` explicitly.

Reply:
[numpy any docment](https://docs.scipy.org/doc/numpy/reference/generated/numpy.any.html)
"Not a Number (NaN), positive infinity and negative infinity evaluate to True because these are not equal to zero."

Reply:
I need to know the difference between CuPy and ClPy. In other words, Why is `nan` treated as `True` in CuPy but not in ClPy? The difference could be a sign of existance of implementation bugs.

Reply:
[opencl docment](https://www.khronos.org/registry/OpenCL/sdk/1.1/docs/man/xhtml/convert_T.html)
"NaN should be converted to 0"

I think this is way I need to add this code

Reply:
OK, I undestand. I think spec of OpenCL is strange... ("NaN is not zero" is right).

Could you write comments to explain the difference?